### PR TITLE
Liveview migration

### LINKED ISSUE #59: Move to LiveView
For example, we could:
1. Order the notebooks by `inserted_at` — the last ones first.

2. Instead of the three buttons on the home page, we could have only one titled "More".
   - When we click, then it would display more notebooks in the table.
   - Then, after the first click on "More", we could have an infinite scroll. That way, when we move below, we always see more notebooks. Or as a first step, we could have the "More" button for now — and leave the infinite scroll for a different issue.

Afterwards, in a different issue, we could add a "Random" button like the one we have right now.

### PR DESCRIPTION
**Pull request by fabrithedev**
Close #59 

Move all features to a liveview
Update all existing controller tests to a liveview tests
Update the router
Remove all old controller files

### CODE REVIEW

**Comment by fabrithedev** [comment]
@hectorperez 
Feel free to msg me if needed.

@BrooklinJazz thkx for indicating this repo. This migration was fun!

File: lib/notesclub_web/live/notesLive.ex [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  import Phoenix.LiveView
+
+  alias NotesclubWeb.NotesLive
+  alias Notesclub.Notebooks
+
+  @notebooks_in_home_count 7
+  def notebooks_in_home_count, do: @notebooks_in_home_count
+
+  def mount(_params, _session, socket) do
+    {:ok, assign(socket, notebooks: [], notebooks_count: 0, search: nil, page: :random)}
```
Comment (hectorperez) [inline lib/notesclub_web/live/notesLive.ex]:
```suggestion
    {:ok, socket}
```
Do we need to assign these variables here?
We're already setting them in `handle_params`.

File: lib/notesclub_web/live/notesLive.ex [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    {:noreply, push_patch(socket, to: Routes.live_path(socket, NotesLive))}
+  end
+
+  def handle_params(params, url, socket) do
+    case connected?(socket) do
+      true ->
+        handle_params_internal(params, url, socket)
+
+      false ->
+        {:noreply, assign(socket, notebooks: [], notebooks_count: 0, search: nil, page: :random)}
```
Comment (hectorperez) [inline lib/notesclub_web/live/notesLive.ex]: Not sure about this as if we don't load notebooks until the socket is connected:
- visitors see a glitch — as there are no notebooks before connecting the socket.
- search engines won't see any notebooks at all (ej. https://notes.club/elixir-nx) — because they don't connect the socket.

To solve this, we could call `handle_params_internal` always — or just have one `handle_params/3 function`.

The homepage/random would still glitch because random notes are different before and after. But this could be less of a problem later when we add an infinite scroll #42 to the home page and move random notebooks to `/random`.

In short, I would delete lines 25-33 and rename `handle_params_internal/3` to `handle_params/3`.
But we could also leave it as it is and change it when we add the infinite scroll.

What do you think?

File: lib/notesclub_web/live/notesLive.ex [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      true ->
+        notebooks = Notebooks.list_random_notebooks(%{limit: @notebooks_in_home_count})
+        count = Notebooks.count()
+
+        {:noreply,
+         assign(socket, notebooks: notebooks, notebooks_count: count, search: nil, page: :random)}
+    end
+  end
+
+  def render(assigns) do
```
Comment (hectorperez) [inline lib/notesclub_web/live/notesLive.ex]: As this is a bit long. I would have put this code in `notes_live.html.heex` instead of `render/1`.
But as you prefer ;)

**Review by hectorperez — COMMENTED** [review]
This is great @FabriDamazio!! 👏👏🏻👏🏼👏🏽👏🏿

I left some comments. Let me know what you think.
Anyway, it's good to be merged! Thanks!!

File: lib/notesclub_web/live/notesLive.ex (hunk 1 above)
Comment (fabrithedev) [reply lib/notesclub_web/live/notesLive.ex]: You are right, we dont need those assignments. Removed

File: lib/notesclub_web/live/notesLive.ex (hunk 2 above)
Comment (fabrithedev) [reply lib/notesclub_web/live/notesLive.ex]: I will remove but we need keep in mind that with the actual implementation when someone access the url directly (like https://notes.club/elixir-nx) the database will be called twice for the same data. But i think we can handle it when needed and avoid early optimizations.

File: lib/notesclub_web/live/notesLive.ex (hunk 3 above)
Comment (fabrithedev) [reply lib/notesclub_web/live/notesLive.ex]: Sure, why not?!

File: lib/notesclub_web/live/notesLive.ex [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    notebooks = Notebooks.list_notebooks(github_filename: term)
+    count = Notebooks.count()
+
+    {:noreply,
+     assign(socket, notebooks: notebooks, notebooks_count: count, search: term, page: :all)}
+  end
+
+  def handle_params(%{}, url, socket) do
+    cond do
+      String.ends_with?(url, "/last_week") ->
```
Comment (hectorperez) [inline lib/notesclub_web/live/notesLive.ex]: This and the other conditions could be replaced by their own functions via pattern matching with `socket.assigns.live_action`:

```elixir
defp handle_params(%{}, url, %{assigns: %{live_action: :last_week}} = socket) do
```

So no need of `String.ends_with?(url, "/last_week")` and similar conditions.

If you prefer me to change it, let me know. You've already done great work!
Or let me know too if you think it's better as it is. Thanks!

**Review by hectorperez — COMMENTED** [review]
These changes look awesome!

I left a comment with a proposal to refactor.
But you can leave it as it is if you prefer and I can do it. Totally ok!
Let me know if you'd prefer me to do it.

Thank you, @FabriDamazio!

File: lib/notesclub_web/live/notesLive.ex (hunk 4 above)
Comment (fabrithedev) [reply lib/notesclub_web/live/notesLive.ex]: @hectorperez if you can do it i'll appreciate because i can't do it today and tomorrow is a day off here in Brazil and probably i wont be online.

I can't wait to help to improve Notesclub. It was a pleasure!

Comment (hectorperez) [reply lib/notesclub_web/live/notesLive.ex]: @FabriDamazio I can do it on Saturday and merge the PR.

The pleasure’s mine!

Comment (fabrithedev) [reply lib/notesclub_web/live/notesLive.ex]: If i can do before saturday i msg here

Comment (fabrithedev) [reply lib/notesclub_web/live/notesLive.ex]: @hectorperez just made the change, please review it =)

Comment (hectorperez) [reply lib/notesclub_web/live/notesLive.ex]: Looks great!!

**Comment by hectorperez** [comment]
@FabriDamazio It's in production now. Thanks! ❤️❤️
